### PR TITLE
Issue#1215 issue#1201

### LINKED ISSUE #1201: Add Acceptation of Terms of use checkbox and link to it to the user subscription form
"I accept the Terms of use and Privacy policy."

### LINKED ISSUE #1215: Add a snackbar validation when a film is added to the wishlist
With a message "Film added to your wishlist!"

### PR DESCRIPTION
**Pull request by ioaNikas**
- [x] Added snackbar when adding/removing movie from selection.
- [x] Added terms of use checkbox in signup and identity pages.

close #1215
close #1201

### CODE REVIEW

File: libs/auth/src/lib/components/signup-form/signup-form.component.scss [hunk 1]
```diff
@@ -18,6 +18,15 @@ form {
   margin-top: 10px;
 }
 
+mat-checkbox {
+  margin-right: 5px;
+}
+
+a {
+  color: #9eb1ff;
```
Comment (GrandSchtroumpf) [inline libs/auth/src/lib/components/signup-form/signup-form.component.scss]: Why ? Please use color from the theme

File: libs/auth/src/lib/components/signup-form/signup-form.component.ts [hunk 2]
```diff
@@ -14,4 +16,15 @@ export class SignupFormComponent {
...
   public signupForm = new SignupForm();
+  public isTermOfUseChedked = false;
+
+  constructor(private dialog: MatDialog) {}
+
+  public openConditions() {
+    this.dialog.open(PrivacyPageComponent, { maxHeight: '100vh' })
+  }
+
+  public checkTermsOfUse() {
```
Comment (GrandSchtroumpf) [inline libs/auth/src/lib/components/signup-form/signup-form.component.ts]: Name this `toggle****` as you change a boolean value.
Also add comments

File: libs/auth/src/lib/components/signup-form/signup-form.component.ts [hunk 3]
```diff
@@ -14,4 +16,15 @@ export class SignupFormComponent {
   @Output() submited = new EventEmitter();
 
   public signupForm = new SignupForm();
+  public isTermOfUseChedked = false;
+
+  constructor(private dialog: MatDialog) {}
+
+  public openConditions() {
```
Comment (GrandSchtroumpf) [inline libs/auth/src/lib/components/signup-form/signup-form.component.ts]: Add comments

File: libs/auth/src/lib/pages/identity/identity.component.scss [hunk 4]
```diff
@@ -27,6 +27,11 @@ h2, p{
   text-align: center;
 }
 
+a {
+  color: #9eb1ff;
```
Comment (GrandSchtroumpf) [inline libs/auth/src/lib/pages/identity/identity.component.scss]: Same as above

File: libs/auth/src/lib/pages/identity/identity.component.ts [hunk 5]
```diff
@@ -60,4 +65,12 @@ export class IdentityComponent {
       this.snackBar.open(error.message, 'close', { duration: 5000 });
     }
   }
+
+  public openConditions() {
```
Comment (GrandSchtroumpf) [inline libs/auth/src/lib/pages/identity/identity.component.ts]: Why do you need to Repeat Yourself ?

File: libs/movie/src/lib/movie/components/display-list/display-list.component.html [hunk 6]
```diff
@@ -30,8 +30,8 @@
   <ng-container matColumnDef="action">
     <th mat-header-cell *matHeaderCellDef></th>
     <td mat-cell *matCellDef="let element">
-      <ng-container *ngIf="isAddedToWishlist(element.id) | async; else alreadyAdded">
-        <mat-icon (click)="addToWishlist(element, $event)" svgIcon="heart_filled"></mat-icon>
+      <ng-container *ngIf="(isAddedToWishlist(element.id) | async) as added; else alreadyAdded">
+        <mat-icon (click)="addToWishlist(element, $event, added)" svgIcon="heart_filled"></mat-icon>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/display-list/display-list.component.html]: same as above

File: apps/catalog/marketplace/marketplace/src/app/movie/home/home.component.html [hunk 7]
```diff
@@ -24,8 +24,8 @@ <h4>{{ section.subline}} </h4>
                 fxLayoutAlign="end start"
                 class="black-bg-opacity-30 overlay"
                 [routerLink]="['/layout/o/catalog', movie.id]">
-                <ng-container *ngIf="(isAddedToWishlist(movie.id) | async); else alreadyAdded">
-                  <button mat-icon-button (click)="addToWishlist(movie, $event)">
+                <ng-container *ngIf="(isAddedToWishlist(movie.id) | async) as added; else alreadyAdded">
+                  <button mat-icon-button (click)="addToWishlist(movie, $event, added)">
```
[the remark this answers is not in the document]
Comment (ioaNikas) [reply apps/catalog/marketplace/marketplace/src/app/movie/home/home.component.html]: `added` is used as an optional argument in `addToWishList()`. This function is also triggered by the button in the `ng-template` below. If added is true (always the case in the `ng-container`). I use it then to generate the right snackbar message.

I will use two different functions instead so it make more sense.

**Pull request merged**